oci/config.py

```diff
@@ -0,0 +1,46 @@
+from collections import namedtuple
+import ConfigParser
+import os
+
+gerrit_default_host = 'gerrit.ovirt.org'
+jenkins_default_host = 'jenkins.ovirt.org'
+
+
+def load():
+    config_dir = os.path.join(os.path.expanduser('~'), '.config/oci.conf')
+    config = ConfigParser.RawConfigParser(allow_no_value=True)
+
+    with open(config_dir, 'r') as config_file:
+        config.readfp(config_file)
+
+        # set default Jenkins host if not present
+        jenkins_host = jenkins_default_host
+        if config.has_option('jenkins', 'host'):
+            jenkins_host = config.get('jenkins', 'host')
+
+        # set default Gerrit host if not present
+        gerrit_host = jenkins_default_host
+        if config.has_option('gerrit', 'host'):
+            gerrit_host = config.get('gerrit', 'host')
+
+        try:
+            token = config.get('jenkins', 'api_token')
+            user = config.get('jenkins', 'user_id')
+        except ConfigParser.NoOptionError as e:
+            raise Exception(
+                'A setting in the config file is missing: ', e)
+
+    Jenkins = namedtuple('jenkins', 'host, user_name, api_token')
+    Gerrit = namedtuple('gerrit', 'host')
+    Config = namedtuple('config', 'jenkins, gerrit')
+
+    result = Config(
+        jenkins=Jenkins(
+            host=jenkins_host,
+            user_name=user,
+            api_token=token),
+        gerrit=Gerrit(
+            host=gerrit_host
+        ))
+
+    return result
```

Comment:
We don't need the result variable, just:
```python
return Config(...)
```

if we set up default before reading the file, we can do this:

```python
    try:
        return Config(
            gerrit=Gerrit(
                host=config.get("gerrit", "host")
            ),
            jenkins=Jenkins(
                host=config.get("jenkins", "host")),
                user_id=config.get("jenkins", "user_id"),
                api_token=config.get("jenkins", "api_token"),
            )
        )
    except ConfigParser.NoOptionError as e:
       handle missing require configuration...
```

Finally it would be useful to have tests for:
- loading configuration from minimal configuration file (only jeknins user_id and api_token defined)
- loading configuration from file with missing required options
- loading configuration from file with different gerrit and jenkins hosts (will be needed for testing later)

To test loading configuration, it would help if we can override the path to
the config file, so we can write test like this:

```python
    def test_load_minimal(tmpdir):
        cfg_data = """
[jenkins]
user_id = fred
api_token = 12345
"""
         cfg_file = tmpdir.join("oci.conf")
         cfg_file.write(cfg_data)
         
         cfg = config.load(str(cfg_file))
         assert cfg.jenkins.user_id == "fred"
         ...
```

Having config file argument to load will be also useful to support usage like:

    oci -f test.conf system-tests 12345

We will need this for writing functional tests with fake gerrit and jenkins server.

Reply:
One thing I am not sure I understand is what is the tmpdir here?
Will the original file `~/.config/oci.conf` will be overwritten? or should the load() function should be able to read the config file from any location?

In this case the load function should look like this, right?
```python
def load(home_dir='~/.config/'):```
    config_dir = os.path.expanduser(home_dir.join("oci.conf"))
```

Reply:
@Funkabell right, load should provide the option to load config from any path.
If path was not specified, load the default configuration.

This should work:
```python
    def load(path=os.path.expanduser("~/.config/oci.conf")):
        ...
```